tex/chProgramming.tex

```diff
@@ -909,23 +914,26 @@ \subsection{Natural numbers}\index[concept]{natural number}
 separated by the \C{|} symbol.  Optionally one can
 prefix the first pattern matching rule with \C{|}, in order to make each line
 begin with \C{|}. Each pattern matching rule results in a new rewrite
-rule available to the \C{compute} command.
+rule available to the \C{Compute} command.
 \index[concept]{pattern matching}
 All
 the pattern matching rules are tried successively against the input.  The
 patterns may overlap, but the result is given by the first pattern that
 matches.
-For instance with the function \C{three_patterns}, if the input is
+For instance, with the function \C{three_patterns}, if the input is
 \C{2}, in other words \C{0.+1.+1}, the first
 rule cannot match, because this would require that \C{0} matches
-\C{u.+1.+1.+1} and we know that \(0\) is not the successor of any
-natural number; when it comes to the second rule \C{0.+1.+1} matches
+\C{u.+1.+1.+1.+1.+1} and we know that \(0\) is not the successor of any
```

Comment:
I disagree with this change: if 0.+1.+1 has to match with u.+1.+1.+1.+1.+1 then by removing two (.+1) simultaneously on the pattern (u.+1.+1.+1.+1.+1) and (0.+1.+1) we get to the situation where 0 has to match (u.+1.+1.+1), as correctly stated in the initial text.

Reply:
You are right, sorry!!